pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -127,13 +138,14 @@ pub(super) async fn connection_manager_loop_step(
                             Ok(()) => debug!("WAL receiving task finished"),
                             Err(e) => error!("wal receiver task finished with an error: {e:?}"),
                         }
+                        // TODO: can this use the `cancel` token?
                         connection_manager_state.drop_old_connection(false).await;
                     },
                 }
             },
 
             // Got a new update from the broker
-            broker_update = broker_subscription.message() => {
+            broker_update = broker_subscription.message() /* TODO: review cancellation-safety */ => {
```

Comment:
We were relying on cancellation-safety before this PR, so, this shouldn't be a blocker.